lib/hello/orders.ex

```diff
@@ -0,0 +1,234 @@
+defmodule Hello.Orders do
+  @moduledoc """
+  The Orders context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Hello.Repo
+
+  alias Hello.Orders.Order
+  alias Hello.Orders.LineItem
+  alias Hello.ShoppingCart
+
+  #function to complete order
+    def complete_order(%ShoppingCart.Cart{} = cart) do
+    line_items =
+      Enum.map(cart.items, fn item ->
+        %{product_id: item.product_id, price: item.product.price, quantity: item.quantity}
+      end)
+
+    order =
+      Ecto.Changeset.change(%Order{},
+        user_uuid: cart.user_uuid,
+        total_price: ShoppingCart.total_cart_price(cart),
+        line_items: line_items
+      )
+
+    Ecto.Multi.new()
+    |> Ecto.Multi.insert(:order, order)
+    |> Ecto.Multi.run(:prune_cart, fn _repo, _changes ->
```

Comment:
@keliumJU Please, explain how Ecto.Multi work

Reply:
Ecto.Multi is the way to do transactions in phoenix, that is, it allows us to perform a series of actions in the database, if any fails, the changes will be undone.